hpedockerplugin/request_context.py

```diff
@@ -0,0 +1,591 @@
+
+    @staticmethod
+    def _validate_opts(operation, contents, valid_opts, mandatory_opts=None):
+        LOG.info("Validating options for operation '%s'" % operation)
+        if 'Opts' in contents and contents['Opts']:
+            received_opts = contents.get('Opts').keys()
+
+            if mandatory_opts:
+                diff = set(mandatory_opts) - set(received_opts)
+                if diff:
+                    # Print options in sorted manner
+                    mandatory_opts.sort()
+                    msg = "One or more mandatory options %s are missing " \
+                          "for operation %s" % (mandatory_opts, operation)
+                    LOG.error(msg)
+                    raise exception.InvalidInput(reason=msg)
+
+            diff = set(received_opts) - set(valid_opts)
+            if diff:
+                diff = list(diff)
+                diff.sort()
+                msg = "Invalid option(s) %s specified for operation %s. " \
+                      "Please check help for usage." % \
+                      (diff, operation)
+                LOG.error(msg)
+                raise exception.InvalidInput(reason=msg)
+
+
+class FileRequestContextBuilder(RequestContextBuilder):
+    def __init__(self, backend_configs):
+        super(FileRequestContextBuilder, self).__init__(backend_configs)
+
+    def _get_build_req_ctxt_map(self):
+        build_req_ctxt_map = OrderedDict()
+        # If share-dir is specified, file-store MUST be specified
+        build_req_ctxt_map['filePersona,help'] = self._create_help_req_ctxt
+        build_req_ctxt_map['filePersona'] = \
+            self._create_share_req_ctxt
+        # build_req_ctxt_map['persona,cpg'] = \
+        #     self._create_share_req_ctxt
+        # build_req_ctxt_map['persona,cpg,size'] = \
+        #     self._create_share_req_ctxt
+        # build_req_ctxt_map['persona,cpg,size,fpg_name'] = \
+        #     self._create_share_req_ctxt
+        # build_req_ctxt_map['virtualCopyOf,shareName'] = \
+        #     self._create_snap_req_ctxt
+        # build_req_ctxt_map['updateShare'] = \
+        #     self._create_update_req_ctxt
+        return build_req_ctxt_map
+
+    def _create_share_req_params(self, name, options, def_backend_name):
+        LOG.info("_create_share_req_params: Entering...")
+        # import pdb
+        # pdb.set_trace()
+        backend = self._get_str_option(options, 'backend', def_backend_name)
+        config = self._backend_configs.get(backend)
+        if not config:
+            raise exception.InvalidInput(
+                'ERROR: Backend %s is not configured for File Persona'
+                % backend
+            )
+        cpg = self._get_str_option(options, 'cpg', config.hpe3par_cpg[0])
+        fpg = self._get_str_option(options, 'fpg', None)
+
+        # Default share size or quota in MiB which is 4TiB
+        size = self._get_int_option(options, 'size', 4 * 1024 * 1024)
```

Comment:
Shall we accept size in GiB instead of MiB? Leaving a comment to pick up discussion later (Making it similar to volume size parameter)